### PR TITLE
Added coniguration for logging slow queries by type (#203)

### PR DESCRIPTION
**Pull request by milannister**
- queryTypesToLog property's value determines which query types will be
logged by slow query module
- improved README.md => Made a link to coremodules.md more visible

### CODE REVIEW

File: cassandra-diagnostics-core/src/main/java/io/smartcat/cassandra/diagnostics/module/slowquery/SlowQueryModule.java [hunk 1]
```diff
@@ -93,20 +93,23 @@ public void process(Query query) {
...
-            fields.put("client", query.clientAddress());
-            fields.put("statement", query.statement());
-
-            final Measurement measurement = Measurement
-                    .create(service, (double) query.executionTimeInMilliseconds(), query.startTimeInMilliseconds(),
-                            TimeUnit.MILLISECONDS, tags, fields);
-
-            logger.trace("Measurement transformed: {}", measurement);
-            report(measurement);
+            if (config.queryTypesToLog().contains("ALL")
```
Comment (nbozic-zz) [inline cassandra-diagnostics-core/src/main/java/io/smartcat/cassandra/diagnostics/module/slowquery/SlowQueryModule.java]: There is [SlowQueryLogDecider](https://github.com/smartcat-labs/cassandra-diagnostics/blob/eaffb0f6b8ed457b616256a123b71b128d36f22d/cassandra-diagnostics-core/src/main/java/io/smartcat/cassandra/diagnostics/module/slowquery/SlowQueryLogDecider.java) which we use to check if correct table is configured, if correct threshold and so on. It is better place to do it. We should maybe create a ticket to discuss if that is best o before any processing but that is outside of scope. Let us first put everything in one place.

Comment (milannister) [reply cassandra-diagnostics-core/src/main/java/io/smartcat/cassandra/diagnostics/module/slowquery/SlowQueryModule.java]: Clear. Will do in a jiffy.

File: cassandra-diagnostics-core/src/main/java/io/smartcat/cassandra/diagnostics/module/slowquery/SlowQueryLogDecider.java [hunk 2]
```diff
@@ -64,6 +64,11 @@ private boolean executionTimeForLogging(long executionTimeInMilliseconds) {
...
     private boolean typeForLogging(Query query) {
         logger.trace("Checking if query type is for logging.");
+        boolean logAll = slowQueryConfiguration.queryTypesToLog().contains("ALL");
+        boolean queryTypeMatches = slowQueryConfiguration.queryTypesToLog().contains(query.statementType().toString());
+        if (!logAll && !queryTypeMatches) {
+            return false;
+        }
         if (query.statementType() == StatementType.SELECT || query.statementType() == StatementType.UPDATE) {
             return true;
         }
```
Comment (nbozic-zz) [inline cassandra-diagnostics-core/src/main/java/io/smartcat/cassandra/diagnostics/module/slowquery/SlowQueryLogDecider.java]: You should check here if type is for logging as well. 

`return logAll || queryTypeMatches;`

And than you do not need two ifs.

File: cassandra-diagnostics-core/COREMODULES.md [hunk 3]
```diff
@@ -29,9 +29,10 @@ Slow Query module is monitoring execution time of each query and if it is above
 
 You can specify slow query threshold (in ms) and list of keyspace.table_name for logging. All options are optional with default threshold of 25ms and all table logging.
 There is a slow query counter available that periodically reports the count of slow queries over the configured period.   
-Two main options that impact reporting are:
+Three main options that impact reporting are:
 * `slowQueryReportEnabled:` Defines is each slow query will be reported (false by default)
 * `slowQueryCountReportEnabled:` Defines if slow query periodic count will be reported (true by default)
+* `queryTypesToLog` - list of types of the query that will be logged. Possible values that can be passed into the list are: UPDATE, SELECT, ALL.
```
Comment (mgobec) [inline cassandra-diagnostics-core/COREMODULES.md]: Please explicitly define default value in braces

File: cassandra-diagnostics-core/src/main/java/io/smartcat/cassandra/diagnostics/module/slowquery/SlowQueryConfiguration.java [hunk 4]
```diff
@@ -24,6 +25,7 @@
         private static final boolean DEFAULT_SLOW_QUERY_COUNT_REPORT_ENABLED = true;
         private static final int DEFAULT_SLOW_QUERY_COUNT_REPORT_PERIOD = 1;
         private static final String DEFAULT_SLOW_QUERY_COUNT_REPORT_TIMEUNIT = "MINUTES";
+        private static final List<String> DEFAULT_SLOW_QUERY_TYPES_TO_LOG = Arrays.asList("ALL");
```
Comment (mgobec) [inline cassandra-diagnostics-core/src/main/java/io/smartcat/cassandra/diagnostics/module/slowquery/SlowQueryConfiguration.java:28]: Can this be enum?

File: cassandra-diagnostics-core/src/main/java/io/smartcat/cassandra/diagnostics/module/slowquery/SlowQueryConfiguration.java [hunk 5]
```diff
@@ -53,6 +55,11 @@
          * Slow query count reporting time unit.
          */
         public TimeUnit slowQueryCountReportTimeunit = TimeUnit.valueOf(DEFAULT_SLOW_QUERY_COUNT_REPORT_TIMEUNIT);
+
+        /**
+         * Slow query statement types to be logged. Only the types in the list will be logged.
+         */
+        public List<String> queryTypesToLog = DEFAULT_SLOW_QUERY_TYPES_TO_LOG;
```
Comment (mgobec) [inline cassandra-diagnostics-core/src/main/java/io/smartcat/cassandra/diagnostics/module/slowquery/SlowQueryConfiguration.java:62]: Can this be enum?

File: cassandra-diagnostics-core/src/main/java/io/smartcat/cassandra/diagnostics/module/slowquery/SlowQueryConfiguration.java [hunk 6]
```diff
@@ -142,4 +149,13 @@ public long slowQueryCountReportingRateInMillis() {
         return slowQueryCountReportTimeunit().toMillis(slowQueryCountReportPeriod());
     }
 
+    /**
+     * Slow query statement types to be logged. Only the types in the list will be logged.
+     *
+     * @return List of query types to be logged.
+     */
+    public List<String> queryTypesToLog() {
```
Comment (mgobec) [inline cassandra-diagnostics-core/src/main/java/io/smartcat/cassandra/diagnostics/module/slowquery/SlowQueryConfiguration.java:157]: Can this return enum?

File: cassandra-diagnostics-core/COREMODULES.md (hunk 3 above)
Comment (milannister) [reply cassandra-diagnostics-core/COREMODULES.md]: done.

**Pull request merged**